pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +89,54 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	rand.NewSource(time.Now().UnixNano()) //creates a more unique random
```

Comment:
concur with Doug. This isn't crypto so having slightly less deterministic rand shouldn't be an issue.